lua/ttt2/libraries/entspawnscript.lua

```diff
@@ -0,0 +1,1072 @@
+			name = "spawn_ammo_shotgun",
+			var = "AMMO_TYPE_SHOTGUN"
+		}
+	},
+	[SPAWN_TYPE_PLAYER] = {
+		[PLAYER_TYPE_RANDOM] = {
+			material = Material("vgui/ttt/tid/tid_big_player"),
+			name = "spawn_player_random",
+			var = "PLAYER_TYPE_RANDOM"
+		}
+	}
+}
+
+local kindToSpawnType = {
+	[WEAPON_PISTOL] = WEAPON_TYPE_PISTOL,
+	[WEAPON_HEAVY] = WEAPON_TYPE_HEAVY,
+	[WEAPON_NADE] = WEAPON_TYPE_NADE,
+	[WEAPON_EQUIP1] = WEAPON_TYPE_SPECIAL,
+	[WEAPON_EQUIP2] = WEAPON_TYPE_SPECIAL,
+	[WEAPON_ROLE] = WEAPON_TYPE_SPECIAL,
+	[WEAPON_MELEE] = WEAPON_TYPE_MELEE,
+	[WEAPON_CARRY] = WEAPON_TYPE_MELEE,
+}
+
+entspawnscript = entspawnscript or {}
+entspawnscript.editingPlayers = entspawnscript.editingPlayers or {}
+
+if SERVER then
+	---
+	-- @realm server
+	local cvUseWeaponSpawnScript = CreateConVar("ttt_use_weapon_spawn_scripts", "1")
+
+	---
+	-- Checks wether a spawnfile for the currently selected map exists.
+	-- @return boolean Returns true if the spawnn script already exists
+	-- @realm server
+	function entspawnscript.Exists()
+		return fileExists(spawndir .. gameGetMap() .. ".txt", "DATA")
+	end
+
+	---
+	-- Initializes the map and generates all the needed spawn files. Called on first load of a map
+	-- if there is no existing spawn file.
+	-- @warning This can break the weapon spawn files if called at any time after the initial spawn wave.
+	-- @return table A table with the default spawn points provided by the map
+	-- @return table A table with the default settings provided by the map
+	-- @realm server
+	function entspawnscript.InitMap()
+		local mapName = gameGetMap()
+
+		-- read the entities from the map at first
+		local spawnTable = {
+			[SPAWN_TYPE_WEAPON] = map.GetWeaponSpawns(),
+			[SPAWN_TYPE_AMMO] = map.GetAmmoSpawns(),
+			[SPAWN_TYPE_PLAYER] = map.GetPlayerSpawns()
+		}
+
+		-- now check if there is a deprectated ttt weapon spawn script and convert the data to
+		-- the new ttt2 system as well
+		if ents.TTT.CanImportEntities(mapName) then
+			local spawns, settings = ents.TTT.ImportEntities(mapName)
+
+			if settings.replacespawns == 1 then
+				spawnTable = {
+					[SPAWN_TYPE_WEAPON] = {},
+					[SPAWN_TYPE_AMMO] = {},
+					[SPAWN_TYPE_PLAYER] = {}
+				}
+			end
+
+			local wepSpawns, ammoSpawns, plySpawns = map.GetSpawnsFromClassTable(spawns)
+
+			-- add new spawns to existing table
+			for entType, addSpawns in pairs(wepSpawns) do
+				spawnTable[SPAWN_TYPE_WEAPON][entType] = spawnTable[SPAWN_TYPE_WEAPON][entType] or {}
+
+				tableAdd(spawnTable[SPAWN_TYPE_WEAPON][entType], addSpawns)
+			end
+
+			for entType, addSpawns in pairs(ammoSpawns) do
+				spawnTable[SPAWN_TYPE_AMMO][entType] = spawnTable[SPAWN_TYPE_AMMO][entType] or {}
+
+				tableAdd(spawnTable[SPAWN_TYPE_AMMO][entType], addSpawns)
+			end
+
+			for entType, addSpawns in pairs(plySpawns) do
+				spawnTable[SPAWN_TYPE_PLAYER][entType] = spawnTable[SPAWN_TYPE_PLAYER][entType] or {}
+
+				tableAdd(spawnTable[SPAWN_TYPE_PLAYER][entType], addSpawns)
+			end
```

Comment:
Combine them by iterating over all keys in spawnTable. And store wep-, ammo- and plySpawns in the same way. Could make the code cleaner.